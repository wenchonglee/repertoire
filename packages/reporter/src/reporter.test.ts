import { execSync } from "node:child_process";
import { promises } from "node:fs";
import { expect, test } from "vitest";
import RepertoireReporter from "./reporter";

test("reporter instantiates run id and url", async () => {
  const reporter = new RepertoireReporter({ url: "test" });

  expect(reporter.runId).toBeTruthy();
  expect(reporter.url).toBe("test/api/");
  expect(reporter.shouldCreateArtifacts).toBe(false);
});

test("reporter produces matching snapshots", async () => {
  execSync("pnpm playwright test --project=chromium reporter-snapshot.spec.ts", {
    env: {
      ...process.env,
      RUN_ID: "vitest",
    },
  });
  const runStart = await promises.readFile("./test-artifacts/run_start.json", "utf8");
  const runStartJson = JSON.parse(runStart);

  expect(runStartJson).toHaveProperty("startTime");
  delete runStartJson.startTime;
  expect(runStartJson).toMatchSnapshot();

  const runEnd = await promises.readFile("./test-artifacts/run_end.json", "utf8");
  const runEndJson = JSON.parse(runEnd);

  expect(runEndJson).toHaveProperty("endTime");
  delete runEndJson.endTime;
  expect(runEndJson).toMatchSnapshot();

  const testStart = await promises.readFile(
    "./test-artifacts/28b2bf81bad5aac29ce2-25ef8da37777e9cf46d7_start.json",
    "utf8"
  );
  const testStartJson = JSON.parse(testStart);

  expect(testStartJson).toHaveProperty("startTime");
  delete testStartJson.startTime;
  expect(testStartJson).toMatchSnapshot();

  const testEnd = await promises.readFile(
    "./test-artifacts/28b2bf81bad5aac29ce2-25ef8da37777e9cf46d7_end.json",
    "utf8"
  );
  const testEndJson = JSON.parse(testEnd);

  expect(testEndJson).toHaveProperty("endTime");
  delete testEndJson.endTime;
  expect(testEndJson).toMatchSnapshot();
});
