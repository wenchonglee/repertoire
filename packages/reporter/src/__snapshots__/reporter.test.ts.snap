// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`reporter 1`] = `
{
  "projects": [
    "chromium",
  ],
  "runId": "vitest",
  "shardId": 1,
  "tests": [
    {
      "annotations": [],
      "expectedStatus": "passed",
      "fileName": "reporter-snapshot.spec.ts",
      "projectName": "chromium",
      "tags": [],
      "testId": "28b2bf81bad5aac29ce2-25ef8da37777e9cf46d7",
      "timeout": 30000,
      "title": "vitest",
      "titlePath": [
        "",
        "chromium",
        "reporter-snapshot.spec.ts",
        "vitest",
      ],
    },
  ],
  "totalShards": 1,
  "version": "1.34.2",
}
`;

exports[`reporter 2`] = `
{
  "status": "passed",
}
`;

exports[`reporter 3`] = `{}`;

exports[`reporter 4`] = `
{
  "attachments": [],
  "errors": [],
  "expectedStatus": "passed",
  "outcome": "expected",
  "status": "passed",
}
`;

exports[`reporter produces matching snapshots 1`] = `
{
  "projects": [
    "chromium",
  ],
  "runId": "vitest",
  "shardId": 1,
  "tests": [
    {
      "annotations": [],
      "expectedStatus": "passed",
      "fileName": "reporter-snapshot.spec.ts",
      "projectName": "chromium",
      "tags": [],
      "testId": "28b2bf81bad5aac29ce2-25ef8da37777e9cf46d7",
      "timeout": 30000,
      "title": "vitest",
      "titlePath": [
        "",
        "chromium",
        "reporter-snapshot.spec.ts",
        "vitest",
      ],
    },
  ],
  "totalShards": 1,
  "version": "1.34.2",
}
`;

exports[`reporter produces matching snapshots 2`] = `
{
  "status": "passed",
}
`;

exports[`reporter produces matching snapshots 3`] = `{}`;

exports[`reporter produces matching snapshots 4`] = `
{
  "attachments": [],
  "errors": [],
  "expectedStatus": "passed",
  "outcome": "expected",
  "status": "passed",
}
`;
